### ISSUE TITLE
[feature] Annotate articles

### ISSUE THREAD

**Issue by erixtekila** [wallabag/wallabag#1130]
Another wonderful enhancement would be to be able to add annotations to the saved article and retrieve those notes inline, in the pdf or pub files later.

Then, wallabag would outmatch all other readitlater apps.

**Comment by tcitworld** [comment]
Yeah, there are much issues opened relative to this. I change the title to export the annotations too, that's a good idea.

**Comment by tcitworld** [comment]
See  #1038

**Comment by nicosomb** [comment]
@erixtekila: FYI, annotations are available since wallabag v2, thanks to @tcitworld!

**Comment by klischka** [comment]
Any news on the (possible) development of an export function for annotions / highlights?

**Comment by MoriTanosuke** [comment]
I'd like to see a list view of articles and their annotation in the UI too. Like the highlights list on Amazon Kindle, where you can see the list of your books with highlights and browse through them book by book. You can also add/edit the highlight via the list page.

**Comment by pykong** [comment]
> I'd like to see a list view of articles and their annotation in the UI too. Like the highlights list on Amazon Kindle, where you can see the list of your books with highlights and browse through them book by book. You can also add/edit the highlight via the list page.

I second that.
Annotating a text means filtering its important content. 
And we could really use a dedicated UI to access that filtered content.

**Comment by Matoo125** [comment]
any news on this feature? It'd be super-useful to see all my annotations in one view.

**Comment by fishhead108** [comment]
Any news here?

**Comment by dbogdanov** [comment]
I agree that this will be a very useful feature. In my use-case I would love to have a simple export of all text highlights and notes for an article into a text clipboard.

**Comment by thelazyoxymoron** [comment]
Another +1 for this feature. I recently migrated from SQLite to Postgres backend and seeing that the none of my annotations were migrated was definitely a letdown. In the meantime I'm using [Hypothesis](https://web.hypothes.is/) for all my annotation needs, but would instantly switch the moment this is ready in wallabag.

Hope I'm not piling on the already stretched dev shoulders. Not familiar with PHP at all, otherwise would've tried to contribute here.

**Comment by linnabraham** [comment]
I am trying out the readwise service. It can access highlights from different services like instapaper and hypothesis. But wallabag is not supported. If at least this manual export thing was an option, maybe I could somehow live with that.

**Comment by j0k3r** [comment]
Does readwise define the structure of the input to provide?
We can provide an export feature but maybe it's better if we have a supported format / structure.

**Comment by linnabraham** [comment]
> Does readwise define the structure of the input to provide? We can provide an export feature but maybe it's better if we have a supported format / structure.

I got this info from the readwise API page at https://readwise.io/api_deets . Is this what you are looking for? 
This specifies a Highlight CREATE request that requires
```
A JSON object with the key highlights, pointing to an array of objects, each with these keys. 
```
And they provide a table of parameters of which only the `highlight text` seems to be a required one.

**Comment by lapineige** [comment]
I believe it is worth mentioning this issue https://github.com/wallabag/wallabag/issues/4907 about being able to show notations in a public link share, as its implementation might be related to this issue ? (and maybe the readwise API thing ?)

**Comment by davidszp** [comment]
A Readwise export integration would be fantastic.

**Comment by theRAGEhero** [comment]
It would be awesome to be able to export and visualize notes.

**Comment by Alanon202** [comment]
@bepolymathe @davidszp @linnabraham There’s [this](https://github.com/rwxd/wallabag2readwise). Seems to work alright, if you slow down the request rate to match the API’s limits. You can export highlights and annotations directly into Readwise, and from there to something like Notion, Roam, etc. Note that exporting from Readwise to something else requires the paid product.

**Comment by rwxd** [comment]
> There’s [this](https://github.com/rwxd/wallabag2readwise). Seems to work alright, if you slow down the request rate to match the API’s limits.

I've added the Readwise rate limiting to this tool.

If somebody has other ideas or  improvements just open a issue :)

**Comment by anonimno1** [comment]
Hi all,

I would like to push the highlights to my local notes. I do not use Readwise.
Is it possible to dump the highlights using https://github.com/rwxd/wallabag2readwise to some kind of local file, e.g. JSON or text or MD?

**Comment by Alanon202** [comment]
> Hi all,
> 
> I would like to push the highlights to my local notes. I do not use Readwise. Is it possible to dump the highlights using https://github.com/rwxd/wallabag2readwise to some kind of local file, e.g. JSON or text or MD?

I use [wallabag-logseq](https://codeberg.org/strubbl/wallabag-logseq) to export all annotations as a single Markdown file that I then use in Logseq. 

The program is quite flexible and you can use templates to manipulate the export (such as what article metadata alongside the text snippet you want to add, etc.). 

Should you prefer json over Markdown, I think the annotations caching file is in json and that would probably also be useable.

**Comment by boston-black** [comment]
This would be a great feature. The workaroudn via LogSeq-Plugin works, but getting a native text export in app would be much easier and more flexible. Thanks <3

**Comment by xeruf** [comment]
Would love to come back to that inline view suggestion, it's a standard nowadays across many Reader apps to be able to see a list of annotations.
